### ISSUE TITLE
Particles can cross the plasma lens without seeing it, for large timesteps

### ISSUE THREAD

**Issue by RemiLehe** [ECP-WarpX/WarpX#3306]
As mentioned in the title, macroparticles can sometimes cross the plasma lens without being affected. See the following plot obtained with this simulation script:
[inputs_3d_zmax0.txt](https://github.com/ECP-WarpX/WarpX/files/9310034/inputs_3d_zmax0.txt)
![Screen Shot 2022-08-11 at 9 00 46 AM](https://user-images.githubusercontent.com/6685781/184177811-e303c2ec-363a-4820-adcd-603aecb698fc.png)

(This uses the following plotting code:
```python
from openpmd_viewer import OpenPMDTimeSeries, ParticleTracker
ts = OpenPMDTimeSeries('./diags/back_3d/')
pt = ParticleTracker( ts, iteration=0, preserve_particle_index=True )
z_trajectories, x_trajectories, y_trajectories = ts.iterate( ts.get_particle, ['z', 'x', 'y'], select=pt)
plt.plot(z_trajectories[:,::200], x_trajectories[:,::200], 'b', alpha=0.1)
plt.axvline( x=0.0145, color='r' )
plt.axvline( x=0.0155, color='r' )
```

When checking which particles are focused, it seems that this depends essentially on their longitudinal position within the bunch:
![Screen Shot 2022-08-11 at 9 00 06 AM](https://user-images.githubusercontent.com/6685781/184177839-2f1bba26-21e5-4c4d-a821-f12fc10dfcaf.png)

It seems that, in this simulation, the distance traveled in z by a macroparticle during one timestep (1.11e-05 m) is comparable to the length of the lens in the boosted-frame (1.66e-05 m). Thus, considering that the lens and particles are counter-propagating in the boosted-frame, it could be that some macroparticles never fall within the range of the lens.

**Comment by dpgrote** [comment]
Currently, the plasma lens requires particles to land at least one time step in the lens for the field to be applied. This can possibly be fixed by adjusting the residence correction that is done for the lens to account for the case when the particle skips over the lens. A PR will be posted soon.

**Comment by dpgrote** [comment]
This should be fixed by PR #3318

**Comment by dpgrote** [comment]
PR #3318 is merged.
